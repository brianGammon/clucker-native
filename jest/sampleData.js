const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
