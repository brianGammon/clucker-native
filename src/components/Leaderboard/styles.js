import { StyleSheet } from 'react-native';

const row = {
  padding: 4,
  flexDirection: 'row',
  alignItems: 'center',
};

const label = {
  width: 40,
  fontWeight: 'bold',
  color: 'grey',
};

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  statsLabel: {
    fontWeight: 'bold',
    fontSize: 17,
  },
  row,
  headerRow: {
    ...row,
    backgroundColor: '#eee',
  },
  label,
  wideLabel: {
    ...label,
    flex: 1,
  },
  value: {
    width: 40,
    textAlign: 'center',
  },
  wideValue: {
    flex: 1,
  },
});
