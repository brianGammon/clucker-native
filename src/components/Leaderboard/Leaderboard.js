/* @flow */
import React from 'react';
import { View, Text } from 'native-base';
import styles from './styles';
import { type FlockStats, type Chicken } from '../../types';

type Props = {
  mode: 'month' | 'allTime',
  stats: FlockStats,
  chickens: {
    [chickenId: string]: Chicken,
  },
};

const Leaderboard = ({ stats, chickens, mode }: Props) => (
  <View style={styles.container}>
    <Text style={styles.statsLabel}>
      {`${mode === 'allTime' ? 'All Time' : ''} Leaderboard`}
    </Text>
    <View style={[styles.row, styles.headerRow]}>
      <Text style={styles.wideLabel}>Chicken</Text>
      <View>
        <Text style={styles.label}>Eggs</Text>
      </View>
    </View>
    {Object.keys(stats.eggsPerChicken).map(key => (
      <View key={key} style={styles.row}>
        <Text style={styles.wideValue}>
          {chickens[key] ? chickens[key].name : 'Unknown'}
        </Text>
        <Text style={styles.value}>{stats.eggsPerChicken[key]}</Text>
      </View>
    ))}
  </View>
);

export default Leaderboard;
