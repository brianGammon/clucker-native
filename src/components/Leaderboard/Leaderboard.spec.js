import React from 'react';
import Leaderboard from '.';
import sampleData from '../../../jest/test-data/full.2017-10-to-2018-05.json';

describe('Leaderboard', () => {
  const stats = {
    eggsPerChicken: {
      chicken1: 3,
      chicken2: 2,
      unknown: 2,
      chicken3: 1,
    },
  };
  const chickens = sampleData.chickens.data;
  test('renders', () => {
    const wrapper = shallow(
      <Leaderboard stats={stats} chickens={chickens} mode="month" />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders nothing when only bulk eggs logged', () => {
    const wrapper = shallow(
      <Leaderboard stats={{ eggsPerChicken: {} }} chickens={{}} mode="month" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
