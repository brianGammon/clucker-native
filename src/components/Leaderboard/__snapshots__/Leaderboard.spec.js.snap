// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard renders 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Styled(Text)
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
  >
     Leaderboard
  </Styled(Text)>
  <Styled(ViewNB)
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 4,
        },
        Object {
          "alignItems": "center",
          "backgroundColor": "#eee",
          "flexDirection": "row",
          "padding": 4,
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "grey",
          "flex": 1,
          "fontWeight": "bold",
          "width": 40,
        }
      }
    >
      Chicken
    </Styled(Text)>
    <Styled(ViewNB)>
      <Styled(Text)
        style={
          Object {
            "color": "grey",
            "fontWeight": "bold",
            "width": 40,
          }
        }
      >
        Eggs
      </Styled(Text)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="chicken1"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Chicken1
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      3
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="chicken2"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Chicken2
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      2
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="unknown"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Unknown
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      2
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    key="chicken3"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Chicken3
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "textAlign": "center",
          "width": 40,
        }
      }
    >
      1
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
